### PR TITLE
app-doc/audacity: stand-alone HTML reference manual for Audacity

### PR DESCRIPTION
**Pull request by mehw**
Hello,

This is a proof of concept, a stand-alone documentation ebuild for `media-sound/audacity-3.3.3` and `audacity-9999`.

This is a way to install HTML pages without requiring the installation or rebuilt of the main `media-sound/audacity` package.

PS: the `metadata.xml` file is derived from `media-sound/audacity`, an almost _cut and paste_.

### CODE REVIEW

File: app-doc/audacity/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <maintainer type="project">
+    <email>proaudio@gentoo.org</email>
+    <name>Gentoo ProAudio Project</name>
+  </maintainer>
+  <maintainer type="person" proxied="yes">
+    <email>richard@audacityteam.org</email>
```
Comment (thesamesam) [inline app-doc/audacity/metadata.xml:9]: Drop

File: app-doc/audacity/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
```
Comment (thesamesam) [inline app-doc/audacity/metadata.xml:3]: Please add yourself here, per https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide#Proxied_maintainer_in_metadata.xml

File: app-doc/audacity/audacity-9999.ebuild [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+else
+	KEYWORDS="amd64 riscv"
+	S="${WORKDIR}/help"
+	SRC_URI="https://github.com/audacity/audacity-manual/releases/download/v${PV}/audacity-manual-${PV}.tar.gz"
+fi
+
+LICENSE="CC-BY-3.0"
+SLOT="0"
+
+DEPEND="!media-sound/audacity[doc]"
```
Comment (thesamesam) [inline app-doc/audacity/audacity-9999.ebuild:21]: Pure-DEPEND blockers don't mean anything, it needs to be RDEPEND.

But I wonder if we should call it audacity-manual or audacity-doc, despite the category, so people don't end up accidentally emerging the wrong thing. Dunno.

File: app-doc/audacity/audacity-9999.ebuild [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+EAPI=8
+
+DESCRIPTION="HTML reference manual for Audacity"
+HOMEPAGE="https://www.audacityteam.org/"
+
+if [[ ${PV} = 9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/audacity/audacity-manual.git"
+else
+	KEYWORDS="amd64 riscv"
```
Comment (thesamesam) [inline app-doc/audacity/audacity-9999.ebuild:13]: Don't commit as stable - ~amd64 and just drop riscv. We can rekeyword it later. Put it under SRC_URI too.

File: app-doc/audacity/metadata.xml (hunk 1 above)
Comment (mehw) [reply app-doc/audacity/metadata.xml]: Done.

File: app-doc/audacity/metadata.xml (hunk 2 above)
Comment (mehw) [reply app-doc/audacity/metadata.xml:3]: I added myself.

File: app-doc/audacity/audacity-9999.ebuild (hunk 3 above)
Comment (mehw) [reply app-doc/audacity/audacity-9999.ebuild]: The repo is https://github.com/audacity/audacity-manual, so maybe `app-doc/audacity-manual` could be a good candidate.

File: app-doc/audacity/audacity-9999.ebuild (hunk 4 above)
Comment (mehw) [reply app-doc/audacity/audacity-9999.ebuild]: Both changes done.

**Comment by mehw** [comment]
@thesamesam thanks for the fixes.

File: app-doc/audacity/audacity-9999.ebuild [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+DESCRIPTION="HTML reference manual for Audacity"
+HOMEPAGE="https://www.audacityteam.org/"
+
+if [[ ${PV} = 9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/audacity/audacity-manual.git"
+else
+	KEYWORDS="~amd64"
+	S="${WORKDIR}/help"
+	SRC_URI="amd64? ( https://github.com/audacity/audacity-manual/releases/download/v${PV}/audacity-manual-${PV}.tar.gz )"
```
Comment (thesamesam) [inline app-doc/audacity/audacity-9999.ebuild:15]: I wonder if this part shouldn't be unconditional. Is anything about it amd64-specific?

(That doesn't mean we should add keywords for other arches until it's been tested, but I'm wondering if the distfile is actually amd64-specific.)

**Review by thesamesam — COMMENTED** [review]
Thanks! One small comment wrt SRC_URI and one request: mind adding yourself as a maint to audacity proper too? 🙏

**Comment by fordfrog** [comment]
@mehw please also add proaudio@gentoo.org project as the mainainer so that i can keep track of this package.

**Comment by mehw** [comment]
> @mehw please also add [proaudio@gentoo.org](mailto:proaudio@gentoo.org) project as the mainainer so that i can keep track of this package.

@fordfrog `metadata.xml` amended.

**Comment by mehw** [comment]
> Thanks! One small comment wrt SRC_URI and one request: mind adding yourself as a maint to audacity proper too? 🙏

@thesamesam as requested https://github.com/gentoo/gentoo/pull/33155.

File: app-doc/audacity/audacity-9999.ebuild (hunk 5 above)
Comment (mehw) [reply app-doc/audacity/audacity-9999.ebuild:15]: After a quick grep operation over the manual, I can't find anything architecture specific except that [Audacity will not run on the Windows RT operating system which ships on devices running ARM architecture](https://manual.audacityteam.org/man/faq_about_audacity.html#mobile).

**Comment by mehw** [comment]
Amended, and also add `audacity-3.4.2`: #35414

**Comment by thesamesam** [comment]
Thanks, I've merged https://github.com/gentoo/gentoo/pull/35414 and then this on top (as I wasn't sure which order to go in and that one had the split commits-per-pkg).

**Comment by mehw** [comment]
@thesamesam thanks for your patience and dedication ;)

**Pull request closed without merging**